django/db/models/sql/query.py

```diff
@@ -1936,7 +1959,11 @@ def resolve_ref(self, name, allow_joins=True, reuse=None, summarize=False):
                     annotation = self.try_transform(annotation, transform)
                 return annotation
             join_info = self.setup_joins(
-                field_list, self.get_meta(), self.get_initial_alias(), can_reuse=reuse
+                field_list,
+                self.get_meta(),
+                self.get_initial_alias(),
+                can_reuse=reuse,
+                reuse_with_filtered_relation=True,
```

Comment:
Ditto about systematically taking the logic meant to only run when resolving filtered relations.